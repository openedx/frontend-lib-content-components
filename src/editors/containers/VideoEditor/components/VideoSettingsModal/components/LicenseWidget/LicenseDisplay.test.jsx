import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';

import { LicenseDisplay } from './LicenseDisplay';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useContext: jest.fn(() => ({ license: ['error.license', jest.fn().mockName('error.setLicense')] })),
}));

describe('LicenseDisplay', () => {
  const props = {
    license: 'all-rights-reserved',
    details: {},
    licenseDescription: 'FormattedMessage component with license description',
    level: 'course',
  };

  describe('snapshots', () => {
    test('snapshots: renders as expected with default props', () => {
      expect(
        shallow(<LicenseDisplay {...props} />).snapshot,
      ).toMatchSnapshot();
    });
    test('snapshots: renders as expected with level set to library', () => {
      expect(
        shallow(<LicenseDisplay {...props} level="library" />).snapshot,
      ).toMatchSnapshot();
    });
    test('snapshots: renders as expected with level set to block', () => {
      expect(
        shallow(<LicenseDisplay {...props} level="block" />).snapshot,
      ).toMatchSnapshot();
    });
    test('snapshots: renders as expected with level set to block and license set to select', () => {
      expect(
        shallow(<LicenseDisplay {...props} level="block" license="select" />).snapshot,
      ).toMatchSnapshot();
    });
    test('snapshots: renders as expected with level set to block and license set to Creative Commons', () => {
      expect(
        shallow(<LicenseDisplay {...props} level="block" license="creative-commons" />).snapshot,
      ).toMatchSnapshot();
    });
  });
});
