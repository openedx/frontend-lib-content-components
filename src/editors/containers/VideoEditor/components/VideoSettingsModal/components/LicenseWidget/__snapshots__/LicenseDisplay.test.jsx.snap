// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LicenseDisplay snapshots snapshots: renders as expected with default props 1`] = `
<Stack
  gap={3}
>
  <div
    className="x-small"
  >
    <FormattedMessage
      defaultMessage="License Display"
      description="Title for license display subsection"
      id="authoring.videoeditor.license.displaySubsection.title"
    />
  </div>
  <div
    className="small border border-gray-300 rounded p-4"
  >
    <injectIntl(ShimmedIntlComponent)
      details={Object {}}
      license="all-rights-reserved"
    />
    <div
      className="x-small mt-3"
    >
      FormattedMessage component with license description
    </div>
  </div>
  <Hyperlink
    className="text-primary-500 x-small"
    destination="https://creativecommons.org/about"
    target="_blank"
  >
    <FormattedMessage
      defaultMessage="View license details"
      description="Label for view license details button"
      id="authoring.videoeditor.license.viewLicenseDetailsLabel.label"
    />
  </Hyperlink>
</Stack>
`;

exports[`LicenseDisplay snapshots snapshots: renders as expected with level set to block 1`] = `
<Stack
  gap={3}
>
  <div
    className="x-small"
  >
    <FormattedMessage
      defaultMessage="License Display"
      description="Title for license display subsection"
      id="authoring.videoeditor.license.displaySubsection.title"
    />
  </div>
  <div
    className="small border border-gray-300 rounded p-4"
  >
    <injectIntl(ShimmedIntlComponent)
      details={Object {}}
      license="all-rights-reserved"
    />
    <div
      className="x-small mt-3"
    >
      FormattedMessage component with license description
    </div>
  </div>
  <Hyperlink
    className="text-primary-500 x-small"
    destination="https://creativecommons.org/about"
    target="_blank"
  >
    <FormattedMessage
      defaultMessage="View license details"
      description="Label for view license details button"
      id="authoring.videoeditor.license.viewLicenseDetailsLabel.label"
    />
  </Hyperlink>
</Stack>
`;

exports[`LicenseDisplay snapshots snapshots: renders as expected with level set to block and license set to Creative Commons 1`] = `
<Stack
  gap={3}
>
  <div
    className="x-small"
  >
    <FormattedMessage
      defaultMessage="License Display"
      description="Title for license display subsection"
      id="authoring.videoeditor.license.displaySubsection.title"
    />
  </div>
  <div
    className="small border border-gray-300 rounded p-4"
  >
    <injectIntl(ShimmedIntlComponent)
      details={Object {}}
      license="creative-commons"
    />
    <div
      className="x-small mt-3"
    >
      FormattedMessage component with license description
    </div>
  </div>
  <Hyperlink
    className="text-primary-500 x-small"
    destination="https://creativecommons.org/about"
    target="_blank"
  >
    <FormattedMessage
      defaultMessage="View license details"
      description="Label for view license details button"
      id="authoring.videoeditor.license.viewLicenseDetailsLabel.label"
    />
  </Hyperlink>
</Stack>
`;

exports[`LicenseDisplay snapshots snapshots: renders as expected with level set to block and license set to select 1`] = `null`;

exports[`LicenseDisplay snapshots snapshots: renders as expected with level set to library 1`] = `
<Stack
  gap={3}
>
  <div
    className="x-small"
  >
    <FormattedMessage
      defaultMessage="License Display"
      description="Title for license display subsection"
      id="authoring.videoeditor.license.displaySubsection.title"
    />
  </div>
  <div
    className="small border border-gray-300 rounded p-4"
  >
    <injectIntl(ShimmedIntlComponent)
      details={Object {}}
      license="all-rights-reserved"
    />
    <div
      className="x-small mt-3"
    >
      FormattedMessage component with license description
    </div>
  </div>
  <Hyperlink
    className="text-primary-500 x-small"
    destination="https://creativecommons.org/about"
    target="_blank"
  >
    <FormattedMessage
      defaultMessage="View license details"
      description="Label for view license details button"
      id="authoring.videoeditor.license.viewLicenseDetailsLabel.label"
    />
  </Hyperlink>
</Stack>
`;
